pkg/controller/appdefinition/serviceaccount.go

```diff
@@ -11,43 +11,45 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func addServiceAccount(appInstance *v1.AppInstance, resp router.Response) {
-	resp.Objects(toServiceAccount(appInstance)...)
-}
-
-func toRules(rules []v1.PolicyRule) (result []rbacv1.PolicyRule) {
-	for _, rule := range rules {
-		result = append(result, (rbacv1.PolicyRule)(rule))
-	}
-	return
-}
-func toServiceAccount(appInstance *v1.AppInstance) (result []kclient.Object) {
-	labelMap := labels.Merge(labels.Managed(appInstance), labels.GatherScoped("", "", appInstance.Status.AppSpec.Labels,
-		nil, appInstance.Spec.Labels))
+func addServiceAccounts(appInstance *v1.AppInstance, resp router.Response) {
+	labelMap := labels.Merge(labels.Managed(appInstance), labels.GatherScoped("", "", appInstance.Status.AppSpec.Labels, nil, appInstance.Spec.Labels))
 	annotations := labels.GatherScoped("", "", appInstance.Status.AppSpec.Annotations, nil, appInstance.Spec.Annotations)
 
-	result = append(result, &corev1.ServiceAccount{
+	// Default "acorn" ServiceAccount to be used by containers without defined permissions
```

Comment:
The `acorn` SA shouldn't exist anymore and we need the SA for the containers/jobs

Reply:
whoops, my bad, i think i told him to do that. So, in the case of no permissions for a particular container, we should still create a service account matching the container name, just dont attach any perms to it, right?

Reply:
The most recent PR updates the logic to be this. To reiterate:

1. When we create a router, job, or container also create a ServiceAccount named after them that those spec use.
2. If the server returned back permissions in `appInstance.Spec.Permissions`, check for the service in those permissions by `serviceName`.
3. If the service has permissions defined, create a the relevant `Roles`/`Bindings`.

> **Note**: Points 2 and 3 do not apply for routers since they cannot have permissions defined.